SharedMemory/RpcBuffer.cs

```diff
@@ -630,28 +710,8 @@ public RpcResponse RemoteRequest(byte[] args = null, int timeoutMs = defaultTime
             ThrowIfDisposedOrShutdown();
 
             var request = CreateMessageRequest();
-            var t = new Task(async () =>
-            {
-                await SendMessage(request, args, timeoutMs).ConfigureAwait(false);
-            });
-            t.Start();
-            
-            if (!request.ResponseReady.WaitOne(timeoutMs))
-            {
-                // Timed out
-                if (request.IsSuccess)
-                {
-                    return new RpcResponse(request.IsSuccess, request.Data);
-                }
-                else
-                {
-                    return new RpcResponse(false, null);
-                }
-            }
-            else
-            {
-                return new RpcResponse(request.IsSuccess, request.Data);
-            }
+            Task<RpcResponse> sendMessage = SendMessage(request, args, timeoutMs);
+            return sendMessage.Result;
```

Comment:
Can you clarify why not async await in this particular instance? I generally avoid task.Result due to the potential to introduce deadlocks.

Reply:
Why not async await 
if i understand the method correct this is the blocking/sync version of RemoteRequest this is why i can use async await

At this place for my point of view an deadlock can never happen because an deadlock only happen if the task need to switch back to the caller conext. Because i have removed all the context switching code (await) deadlock will never happen